### PR TITLE
Fix CSettingControlList regression after adding user values option

### PR DESCRIPTION
**Pull request by DaveTBlake**
Follow up to https://github.com/xbmc/xbmc/pull/18236 CSettingControlList  - the addition of option to allow users to enter new values not on the preset list, which caused a regression in some types of selection list setting controls. Reported here https://github.com/xbmc/xbmc/pull/18236#issuecomment-671101025

The problem was casting setting as `CSettingString` to get property, without conditional checks for other list element types, and using `GetDefinition()`

@ksooo thanks for picking up on this so quickly

### CODE REVIEW

File: xbmc/settings/lib/Setting.h [hunk 1]
```diff
@@ -402,7 +402,7 @@ class CSettingString : public CTraitedSetting<std::string, SettingType::String>
 
   virtual bool AllowEmpty() const { return m_allowEmpty; }
   void SetAllowEmpty(bool allowEmpty) { m_allowEmpty = allowEmpty; }
-  bool AllowNewOption() const { return m_allowNewOption; }
+  virtual bool AllowNewOption() const { return m_allowNewOption; }
```
Comment (ksooo) [inline xbmc/settings/lib/Setting.h:405]: Why this change?

Comment (DaveTBlake) [reply xbmc/settings/lib/Setting.h:405]: As originally intended (like AllowEmpty) but somehow got lost. I spotted it when checking the rest so slipped in a one word fix rather than yet another PR

Comment (ksooo) [reply xbmc/settings/lib/Setting.h:405]: Okay.

**Review by ksooo — APPROVED** [review]
LGTM, Thx.
